go/runtime/registry/host_rofl_mgmt.go

```diff
@@ -68,21 +91,44 @@ func (rh *roflHostHandler) handleBundleWrite(rq *rofl.BundleWriteRequest) (*rofl
 	return &rofl.BundleWriteResponse{}, nil
 }
 
-func (rh *roflHostHandler) handleBundleAdd(rq *rofl.BundleAddRequest) (*rofl.BundleAddResponse, error) {
+func (rh *roflHostHandler) handleBundleAdd(ctx context.Context, rq *rofl.BundleAddRequest) (*rofl.BundleAddResponse, error) {
 	if err := rh.ensureComponentPermissions(runtimeConfig.PermissionBundleAdd); err != nil {
 		return nil, err
 	}
 
+	select {
+	case <-rh.getVolumeManager().Initialized():
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	}
+
 	// Determine labels, make sure to override origin as that is used for isolation.
 	labels := maps.Clone(rq.Labels)
 	maps.Copy(labels, rh.getBundleManagementLabels())
 
+	// Resolve volumes.
+	volumes := make(map[string]*volume.Volume)
+	for volName, volID := range rq.Volumes {
+		volume, err := rh.getVolumeManager().Get(volume.ID(volID))
+		if err != nil {
+			return nil, err
+		}
+
+		// Ensure volume has the right labels.
+		if !volume.HasLabels(rh.getBundleManagementLabels()) {
```

Comment:
Creating labels for every volume could be avoided. Also, do we need to check if the volume also contains the request labels? Is this verification needed, why would volume with unique ID change its labels?

Reply:
This is a defense in depth check in case the scheduler would guess a volume ID that it doesn't have access to.